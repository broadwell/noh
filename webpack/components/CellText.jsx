import React from "react";
import PropTypes from "prop-types";

const CellText = props => {
  let rangeSpan = null;
  let rangeClass = "";
  if (props.vocalRange === "l") {
    rangeClass = "low";
  } else if (props.vocalRange === "m") {
    rangeClass = "medium";
  } else if (props.vocalRange === "h") {
    rangeClass = "high";
  }
  if (props.vocalRange !== "") {
    rangeSpan = <span className={`cell__range--${rangeClass}`} />;
  } else {
    rangeSpan = null;
  }
  let typeSpan = null;
  let tooltipSpan = null;
  if (props.voiceType !== "") {
    typeSpan = <span className={`cell__character--${props.voiceType}`} />;
    tooltipSpan = (
      <span
        className={`cell__tooltip${
          props.textIsCongruent ? " cell__tooltip--centered" : ""
        }`}
      >
        {props.voiceType}
      </span>
    );
  }
  return (
    <div
      className={`cell cell--${props.length} cell--text${
        props.voiceType ? " cell__character" : ""
      }`}
    >
      {props.textIsCongruent ? (
        props.text
      ) : (
        <span className="truncate">{props.text}</span>
      )}
      {typeSpan}
      {tooltipSpan}
      {rangeSpan}
    </div>
  );
};

CellText.propTypes = {
  text: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired,
  vocalRange: PropTypes.string,
  voiceType: PropTypes.string,
  textIsCongruent: PropTypes.bool
};

CellText.defaultProps = {
  vocalRange: "",
  voiceType: "",
  textIsCongruent: false
};

export default CellText;
