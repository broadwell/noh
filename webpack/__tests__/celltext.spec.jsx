import React from "react";
import { shallow } from "enzyme";
import CellText from "../components/CellText";

describe("<CellText>", () => {
  it("renders as expected", () => {
    const component = shallow(
      <CellText
        text="ma"
        length={2}
        voiceType="jiutai"
        vocalRange="high"
        beginning
        end
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders as expected with no vocal range", () => {
    const component = shallow(
      <CellText text="ma" length={2} voiceType="jiutai" beginning end />
    );
    expect(component).toMatchSnapshot();
  });
});
