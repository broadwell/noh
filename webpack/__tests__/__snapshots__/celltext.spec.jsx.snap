// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CellText> renders as expected 1`] = `
<div
  className="cell cell--2 cell--text"
>
  ma
  <span
    className="cell__character cell__character--jiutai cell__character--begin cell__character--end"
  />
  <span
    className="cell__tooltip"
  >
    jiutai
  </span>
  <span
    className="cell__range--high"
  />
</div>
`;

exports[`<CellText> renders as expected with no vocal range 1`] = `
<div
  className="cell cell--2 cell--text"
>
  ma
  <span
    className="cell__character cell__character--jiutai cell__character--begin cell__character--end"
  />
  <span
    className="cell__tooltip"
  >
    jiutai
  </span>
</div>
`;
